### PR TITLE
Make flaw comments independent of Bugzilla

### PR DESCRIPTION
**Pull request by dmonzonis**
This commit redesigns the FlawComment model as well as the API to make them independent of Bugzilla sync. Before this commit, the comment info was being stored in meta_attr, which is filled during the sync with Bugzilla. Some info was also shared in the base model (text) but wasn't being sent through the API, while other info was completely lost (creator, is_private, time).

With this commit, the info is stored in the model, and during sync it is updated. The info in the model is also serialized and sent via the API.

Closes OSIDB-2760.

### CODE REVIEW

File: osidb/models.py [hunk 1]
```diff
@@ -2988,6 +2996,14 @@ class Meta:
             GinIndex(fields=["acl_read"]),
         ]
 
+    @property
+    def time(self):
+        """
+        Return the time of the BZ comment if sync is enabled, or the
+        record's creation datetime otherwise.
+        """
+        return self.meta_attr.get("time", self.created_dt)
```
Comment (osoukup) [inline osidb/models.py:3005]: I am thinking why is this necessary. Wouldn't it be better to just set `created_dt` to the Bugzilla `time` value in the `FlawConvertor` if fetched from Bugzilla and just leave it to be automatically set by the `TrackingMixin` otherwise? When/why should they be different?

**Review by osoukup — CHANGES_REQUESTED** [review]
Mostly LGTM but I raised a few points to answer/address.

File: osidb/models.py (hunk 1 above)
Comment (dmonzonis) [reply osidb/models.py:3005]: I did that because `meta_attr.time` is what OSIM checks to make the transition more clear, but thinking it twice, I think you're right since we shouldn't bend OSIDB to adhere to OSIM standards, but the other way around. So I will definitely change that.

File: osidb/migrations/0132_flawcomment_redesign.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    set_user_acls(settings.ALL_GROUPS)
+    FlawComment = apps.get_model("osidb", "FlawComment")
+
+    # Move existing comment data from meta_attr to the model
+    comments = FlawComment.objects.all().iterator(chunk_size=BATCH_SIZE)
+
+    batch = []
+    for i, comment in enumerate(comments, 1):
+        comment.creator = comment.meta_attr.get("creator", "")
+        # is_private seems to be stored as a string in old data
```
Comment (JakubFrejlach) [inline osidb/migrations/0132_flawcomment_redesign.py:20]: Note: This is not entirely true. `meta_attr` is hstore field type and such field can only hold key:values records where each value is a string, meta_attr simply cannot hold any other type. I believe in old data, there is a boolean in place.

**Review by JakubFrejlach — APPROVED** [review]
LGTM

File: osidb/migrations/0132_flawcomment_redesign.py (hunk 2 above)
Comment (dmonzonis) [reply osidb/migrations/0132_flawcomment_redesign.py:20]: I see. I will check for both types then just in case. Inconsistencies like this are yet another good reason to make the switch.

**Review by osoukup — APPROVED** [review]
LGTM
